### PR TITLE
Math primitives cleanup

### PR DESCRIPTION
**Pull request by solis-lumine-vorago**
# Objective

- General clenup of the primitives in `bevy_math`
- Add `eccentricity()` to `Ellipse`

## Solution

- Moved `Bounded3d` implementation for `Triangle3d` to the `bounded` module
- Added `eccentricity()` to `Ellipse`
- `Ellipse::semi_major()` and `::semi_minor()` now accept `&self` instead of `self`
- `Triangle3d::is_degenerate()` actually uses `f32::EPSILON` as documented
- Added tests for `Triangle3d`-maths

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

**Review by Jondolf — COMMENTED** [review]
Looks good overall, left a couple of nits

**Review by Jondolf — APPROVED** [review]

File: crates/bevy_math/src/primitives/dim3.rs [hunk 1]
```diff
@@ -774,7 +774,7 @@ impl Triangle3d {
         let [a, b, c] = self.vertices;
         let ab = b - a;
         let ac = c - a;
-        ab.cross(ac).length() < 10e-7
+        ab.cross(ac).length() < EPSILON
```
Comment (NthTensor) [inline crates/bevy_math/src/primitives/dim3.rs:777]: So I will just note that at one point the PR that added this used the machine epsilon, and then it was changed to use 10e-7 (they apparently forgot to update the docs). I have a lingering suspicion I suggested this change. @vitorfhc can you fill us in here?

**Review by NthTensor — CHANGES_REQUESTED** [review]
I'd like to hold off on this, as I'm not sure if the machine epsilon is appropriate. The original PR specifically moved away from this approach. 

Otherwise looks great!

File: crates/bevy_math/src/primitives/dim3.rs (hunk 1 above)
Comment (Jondolf) [reply crates/bevy_math/src/primitives/dim3.rs:777]: Yep, link to that thread here: https://github.com/bevyengine/bevy/pull/12508#discussion_r1534137670

Comment (alice-i-cecile) [reply crates/bevy_math/src/primitives/dim3.rs:777]: Yeah, we shouldn't use machine epsilon here.

**Review by NthTensor — CHANGES_REQUESTED** [review]
Still need to fix the docs so to have the correct epsilon.

**Review by NthTensor — APPROVED** [review]
Tysm & Lgtm! Sorry to make you jump through hoops a bit.

**Pull request merged**